cpp/src/interop/arrow_utilities.hpp

```diff
@@ -86,5 +87,15 @@ std::unique_ptr<rmm::device_buffer> decimals_to_arrow(cudf::column_view input,
                                                       rmm::cuda_stream_view stream,
                                                       rmm::device_async_resource_ref mr);
 
+template <typename T, class Enable = void>
+struct DeviceType {
```

Comment:
There is already (e.g.) cudf::detail::DeviceType defined in copy_if.cuh. Do we really need this?

Reply:
oh nice! I didn't see that one there, we don't need this at all. I'll replace the usages of this with the one from `copy_if.cuh`

Reply:
This is being removed in https://github.com/rapidsai/cudf/pull/17520.
We may want to explore adding a new dispatch-type similar to [`dispatch_storage_type`](https://github.com/rapidsai/cudf/blob/13e983eafecad5a3d4053157febd714e40a410c3/cpp/include/cudf/utilities/type_dispatcher.hpp#L211) perhaps?

Reply:
Just noting that it does not include timestamp `rep` type. 
But that is not needed here so I think we should be good to use `dispatch_storage_type`

Reply:
updated